x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/home/available_packages.tsx

```diff
@@ -52,6 +56,76 @@ import type { CategoryFacet } from './category_facets';
 import type { CategoryParams } from '.';
 import { getParams, categoryExists, mapToCard } from '.';
 
+const NoEprCallout: FunctionComponent<{ statusCode?: number }> = ({
+  statusCode,
+}: {
+  statusCode?: number;
+}) => {
+  let titleMessage;
+  let descriptionMessage;
+  if (statusCode === 500) {
+    titleMessage = i18n.translate('xpack.fleet.epmList.eprUnavailable400500CalloutTitle', {
+      defaultMessage:
+        'Kibana encounters an error connecting to the Elastic Package Registry, which provides Elastic Agent integrations.\n',
+    });
+    descriptionMessage = (
+      <FormattedMessage
+        id="xpack.fleet.epmList.eprUnavailableCallout400500TitleMessage"
+        defaultMessage="Please ensure the {registryproxy} or {onpremregistry} is configured correctly, or try again later."
+        values={{
+          registryproxy: <ProxyLink />,
+          onpremregistry: <OnPremLink />,
+        }}
+      />
+    );
+  } else {
```

Comment:
nit: I think we should make the logic be more specific to `if 502` case to show the callout about configuring EPR differently and the catch all `else` be the more generic callout.